docs/versioned_docs/version-zenith/user/quickstart/842901-test.mdx

```diff
@@ -0,0 +1,73 @@
+---
+slug: /user/quickstart/842901/test
+hide_table_of_contents: true
+title: "Add a function to test the application"
+---
+
+import PartialExperimentalDocs from '../../partials/_experimental.mdx';
+
+# Quickstart
+
+<PartialExperimentalDocs />
+
+## Add a function to test the application
+
+The function in the previous example was illustrative - you wouldn't ever use it in the real world! So, the next step is to write a Dagger function to do something more useful, like running an application's tests.
+
+Since the application is a Node.js application, it's convenient to start with the [`node` module](https://daggerverse.dev/mod/github.com/quartz-technology/daggerverse/node) from the [Daggerverse](https://daggerverse.dev), which provides a set of ready-made functions to manage a Node.js project.
+
+:::info
+The [Daggerverse](https://daggerverse.dev) is an online catalog of Dagger modules that enables users to easily discover and use modules created by the Dagger community.
+:::
+
+Begin by adding the `node` module as a dependency:
+
+```shell
+dagger mod install github.com/quartz-technology/daggerverse/node
+```
+
+Then, add a new `Test()` function to `main.go`, which runs the application's unit test suite:
+
+```go file=snippets/test/main.go
+```
+
+This function does the following:
+
+- It calls the `node` module's `WithVersion()` function via the `dag` client. This function returns a `node` container image with the given Node.js version. This container image is represented as a `Node` object.
+- It calls the module's `Node.WithNpm()` function, which returns a revised `Node` object after adding the `npm` package manager and a cache volume for `npm`.
+- It calls the module's `Node.WithSource()` function, which returns a revised `Node` object including the application source code and a cache volume for Node.js modules. The `dag.Host().Directory()` method queries the host environment and returns a reference to the application source code directory on the host after excluding the `.git` and `node_modules` sub-directories.
```

Comment:
This whole subject is worth its own dedicated guide (in time, not needed in this PR), but we should clarify here that `dag.Host().Directory()` does not read from the CLI caller's filesystem. It reads files from the module container's filesystem.

It works here because when a module is loaded into dagger, it includes the directory tree at which `dagger.json` is located. That means the source code is included in the module container at the current working directory where `dag.Host().Directory()` is executed in that container.

I realize that's a lot to try to explain just here, but we should at least mention it here I think (and then rely on the aforementioned dedicated guide in the long run for explaining the full details).

Reply:
+1

Reply:
Added